### PR TITLE
added delete tags and delete repository commands

### PR DESCRIPTION
**Pull request by doctori**
Hi, 
Thank you for your work on this nice projet.
I needed a few more functionalities like removing tags and repositories (you know for cleaning)
so I added them.
I know this is a bit dirty, but le me know what would need some improvement and I will refine the code.

By the way, the DELETE repository API Call will launch a removal of all the tags before removing the repository, this might induce some timeout since it waits for all the actions to complete.

I've also change the command list-tag to tag-list to be more coherent with the rest of the commands

### CODE REVIEW

File: harborclient/v2/repositories.py [hunk 1]
```diff
@@ -11,11 +11,21 @@ def list(self, project):
         repositories = self._list("/repositories?project_id=%s" % project)
         return repositories
 
+    def delete_repository(self, repo_name):
+        """Delete the repository."""
+        return self.api.client.delete(
+            "/repositories/%s" % repo_name)
+
```
Comment (int32bit) [inline harborclient/v2/repositories.py:18]: I think there may be two blank lines to fit pep8 style.

File: harborclient/v2/shell.py [hunk 2]
```diff
@@ -341,15 +342,96 @@ def do_list(cs, args):
...
+        repositories = cs.repositories.list(cs.client.project)
+        for repository in repositories:
+            if re.match(args.repo_reg,repository['name']):
+                if args.dryrun:
+                    print("Would have removed : %s" % repository['name'])
+                else:
+                    cs.repositories.delete_repository(repository['name'])
+                    print("Repository %s deleted" % repository['name'])
+    elif args.repository:
+        # unique repository removal
```
Comment (int32bit) [inline harborclient/v2/shell.py]: Should we handle `--dry-run` option here ?

File: harborclient/v2/shell.py [hunk 3]
```diff
@@ -341,15 +342,96 @@ def do_list(cs, args):
...
     utils.print_list(tags, fields, sortby="name")
 
 
+@utils.arg('repository', metavar='<repository>', help='Name of repository.')
+@utils.arg('tag', metavar='<tag>', help='Name of the tag.')
+def do_tags_delete(cs, args):
+    """delete tag of a relevant repository."""
+    rc = cs.repositories.delete_tags(args.repository, args.tag)
+
+    print("Delete tag '%s:%s' sucessfullywith return code %s" %
```
Comment (int32bit) [inline harborclient/v2/shell.py]: It's much better if you can catch exception such as `404 Not Found`.

**Review by int32bit — CHANGES_REQUESTED** [review]

File: harborclient/v2/shell.py (hunk 3 above)
Comment (doctori) [reply harborclient/v2/shell.py]: I'm trying to catch the exception, but it seems that the client doesn't raise any exception, and the body is systematically empty. I'll try to have more details about the return of harbor on the DELETE action, but I can't seem to make it failed...  
If you have any idea ?

Comment (doctori) [reply harborclient/v2/shell.py]: Okay FYI Harbor doesn't return 404 on a tag not found but a 404 for a repository not found (with a message `project not found`)

File: harborclient/v2/repositories.py (hunk 1 above)
Comment (doctori) [reply harborclient/v2/repositories.py:18]: the two blank lines would be for methods outside a class.  
inside a class it's only one.  
pycodestyle returns nothing

**Comment by int32bit** [comment]
Thank you @doctori !

**Pull request merged**